### PR TITLE
feat: remove fortmatic

### PR DESCRIPTION
**Pull request by vm**
This should significantly decrease our app's load time.

### CODE REVIEW

File: src/utils/switchChain.ts [hunk 1]
```diff
@@ -41,19 +34,8 @@ function getRpcUrls(chainId: SupportedChainId): [string] {
...
-    case injectedConnection.connector:
-    case coinbaseWalletConnection.connector:
-    case walletConnectConnection.connector:
-    case networkConnection.connector:
-    case gnosisSafeConnection.connector:
-      return ALL_SUPPORTED_CHAIN_IDS.includes(chainId)
-    default:
-      return false
-  }
+export function isChainAllowed(_connector: Connector, _chainId: number) {
```
Comment (vm) [inline src/utils/switchChain.ts:37]: rm this function now

File: src/hooks/useEagerlyConnect.ts [hunk 2]
```diff
@@ -18,15 +19,26 @@ async function connect(connector: Connector) {
...
+  const dispatch = useAppDispatch()
+
   const selectedWalletBackfilled = useAppSelector((state) => state.user.selectedWalletBackfilled)
   const selectedWallet = useAppSelector((state) => state.user.selectedWallet)
 
+  let selectedConnection: Connection | undefined
+  if (selectedWallet) {
+    try {
+      selectedConnection = getConnection(selectedWallet)
+    } catch {
```
Comment (cmcewen) [inline src/hooks/useEagerlyConnect.ts:31]: this is unrelated to the fortmatic removal right? just another improvement

File: src/lib/hooks/useCurrency.ts [hunk 3]
```diff
@@ -105,8 +102,7 @@ export function useCurrencyFromMap(tokens: TokenMap, currencyId?: string | null)
 
   const token = useTokenFromMapOrNetwork(tokens, isNative ? undefined : shorthandMatchAddress ?? currencyId)
 
-  const chainAllowed = chainId && isChainAllowed(connector, chainId)
-  if (currencyId === null || currencyId === undefined || !chainAllowed) return null
+  if (currencyId === null || currencyId === undefined) return null
```
Comment (cmcewen) [inline src/lib/hooks/useCurrency.ts:105]: this also seems unrelated? would be nice to keep the diff clean

File: src/hooks/useEagerlyConnect.ts (hunk 2 above)
Comment (vm) [reply src/hooks/useEagerlyConnect.ts:31]: it is related because if FORTMATIC is the saved option this all will error out

File: src/lib/hooks/useCurrency.ts (hunk 3 above)
Comment (vm) [reply src/lib/hooks/useCurrency.ts:105]: i could have the function just return `true` and leave all this if we wanted to

Comment (vm) [reply src/lib/hooks/useCurrency.ts]: reverted

Comment (vm) [reply src/lib/hooks/useCurrency.ts]: this actually was a bug, i needed to check ALLOWED_CHAIN_IDS.

Comment (vm) [reply src/lib/hooks/useCurrency.ts]: i'd like to rename this to `isChainSupported` separately also

**Review by cmcewen — APPROVED** [review]
LGTM

**Comment by vm** [comment]
Gonna hold off on this until later because need to write up a doc explaining it for the team

**Review by zzmp — APPROVED** [review]
